### PR TITLE
Docs for dynamic credentials with Kubernetes

### PR DESCRIPTION
**Pull request by alexsomesan**
### What

Adding a new page describing how to use dynamic credentials with Kubernetes.

### Why

TFC is about to get support for dynamic credentials with Kubernetes.

https://github.com/hashicorp/atlas/pull/17256

### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [X] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [X] Description links to related pull requests or issues, if any.

#### Content
- [ ] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [ ] API documentation and the API Changelog have been updated. 
- [ ] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [X] Pages with related content are updated and link to this content when appropriate.
- [X] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [ ] New pages have metadata (page name and description) at the top.
- [ ] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [X] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [ ] UI elements (button names, page names, etc.) are bolded.
- [X] The Vercel website preview successfully deployed.

#### Reviews
- [X] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+The value for "issuer URL" should be set to the address of Terraform Cloud (e.g., https://app.terraform.io **without** a trailing slash) or the URL of your Terraform Enterprise instance. The value for "client ID" is your audience in OIDC terminology and it should match the value of the `TFC_KUBERNETES_WORKLOAD_IDENTITY_AUDIENCE` environment variable in your workspace.
+
+The OIDC identity will resolve authentication to the Kubernetes API, but it requires authorization in order to interact with the Kubernetes API. You must bind RBAC roles to the OIDC identity in Kubernetes. You can use both "User" and "Group" subjects in your role bindings. For OIDC identities coming from TFC, the User value will have a format of `organization:<MY-ORG-NAME>:project:<MY-PROJECT-NAME>:workspace:<MY-WORKSPACE-NAME>:run_phase:<plan|apply>`. The Group value is extracted form the token claim configured for this purpose in your cluster OIDC configuration. The complete structure of the TFC token is described here: [Workload Identity](/terraform/cloud-docs/workspaces/dynamic-provider-credentials/workload-identity-tokens)
```
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx:32]: This was a lot of information at once, so I suggest splitting this up a bit for readability, like so:
```suggestion
The OIDC identity resolves authentication to the Kubernetes API, but it first requires authorization to interact with that API. So, you must bind RBAC roles to the OIDC identity in Kubernetes.

You can use both "User" and "Group" subjects in your role bindings. For OIDC identities coming from TFC, the "User" value is formatted like so: `organization:<MY-ORG-NAME>:project:<MY-PROJECT-NAME>:workspace:<MY-WORKSPACE-NAME>:run_phase:<plan|apply>`. 

You can extract the "Group" value from the token claim you configured in your cluster OIDC configuration. For details on the structure of the TFC token, refer to [Workload Identity](/terraform/cloud-docs/workspaces/dynamic-provider-credentials/workload-identity-tokens).
```

Comment (rkoron007) [reply website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx:32]: To add to the above, could we put `organization:<MY-ORG-NAME>:project:<MY-PROJECT-NAME>:workspace:<MY-WORKSPACE-NAME>:run_phase:<plan|apply>` in a code block? The long string is hard to follow in the middle of the paragraph.

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+A `RoleBinding` for the Terraform Cloud OIDC identity will looks like this:
```
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx:34]:
```suggestion
Below, we show an example of a `RoleBinding` for the Terraform Cloud OIDC identity.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+  subject {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "User"
+    name      = "organization:${var.tfc_organization_name}:project:${var.tfc_project_name}:workspace:${var.tfc_workspace_name}:run_phase:apply"
+  }
+}
+```
+
+When using User subjects for the binding, be aware that Plan and Apply phases get assigned different identities and they each need specific bindings. This way you can tailor the permissions for each phase. Plans usually only requires read-only permissions while applies also requires write access.
```
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx:78]:
```suggestion
If binding with "User" subjects, be aware that plan and apply phases are assigned different identities, each requiring specific bindings. Meaning you can tailor permissions for each Terraform operation. Planning operations usually require "read-only" permissions, while apply operations also require "write" access.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+!> **Warning**: you should always check, at minimum, the audience and the name of the organization in order to prevent unauthorized access from other Terraform Cloud organizations!
```
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx:80]:
```suggestion
!> **Warning**:  Always check, at minimum, the audience and the organization's name to prevent unauthorized access from other Terraform Cloud organizations.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+## Configure Terraform Cloud
+
+You’ll need to set some environment variables in your Terraform Cloud workspace in order to configure Terraform Cloud to authenticate with AWS using dynamic credentials. You can set these as workspace variables, or if you’d like to share one AWS role across multiple workspaces, you can use a variable set.
```
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx:84]: I think this might have been copy pasta? I'm not sure what the last sentence's equivalent would be for k8 and Helm though? 
```suggestion
You must set certain environment variables in your Terraform Cloud workspace to configure Terraform Cloud to authenticate with Kubernetes and Helm using dynamic credentials. You can set these as workspace variables, or if you’d like to share one AWS role across multiple workspaces, you can use a variable set.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+| TFC_KUBERNETES_PROVIDER_AUTH | `true` | Requires **v1.14.0** or later if self-managing agents. Must be present and set to `true`, or Terraform Cloud will not attempt to authenticate to Kubernetes. |
+| TFC_KUBERNETES_WORKLOAD_IDENTITY_AUDIENCE | The audience name in your cluster's OIDC configuration, such as `kubernetes`. | Requires **v1.14.0** or later if self-managing agents. |
+
+## Configure the provider
+
+The Kubernetes and Helm providers share the same schema of configuration attributes for the provider block. The example below illustrates using the Kubernetes provider but the same configuration applies to the Helm provider.
+
+Make sure that you are not using any of the other arguments or methods listed in the [authentication](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#authentication) section of the provider documentation as these settings may interfere with dynamic provider credentials. The only allowed provider attributes `host` and `cluster_ca_certificate`.
```
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx:96]:
```suggestion
Make sure that you are not using any of the other arguments or methods listed in the [authentication](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#authentication) section of the provider documentation as these settings may interfere with dynamic provider credentials. The only allowed provider attributes are `host` and `cluster_ca_certificate`.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+### Single provider instance
+
+Terraform Cloud will automatically set the `KUBE_TOKEN` environment variable and include the workload identity token.
```
Comment (rkoron007) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx:100]:
```suggestion
Terraform Cloud automatically sets the `KUBE_TOKEN` environment variable and includes the workload identity token.
```

**Review by rkoron007 — APPROVED** [review]
Looks great to me! Added some smaller suggestions for style, but please feel free to push back on anything that doesn't feel good to you!

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+### Configure an OIDC Identity Provider
+
+AWS documentation describes setting up an EKS cluster for OIDC authentication here: [Authenticating users for your cluster from an OpenID Connect identity provider](https://docs.aws.amazon.com/eks/latest/userguide/authenticate-oidc-identity-provider.html). Refer to our [example Terraform configuration](https://github.com/hashicorp-education/learn-terraform-dynamic-credentials/tree/main/eks/trust).
+
+GCP documentation describes setting up a GKE cluster for OIDC authentication here: [Use external identity providers to authenticate to GKE](https://cloud.google.com/kubernetes-engine/docs/how-to/oidc). Refer to our [example Terraform configuration](https://github.com/hashicorp-education/learn-terraform-dynamic-credentials/tree/main/gke/trust).
+
+The value for "issuer URL" should be set to the address of Terraform Cloud (e.g., https://app.terraform.io **without** a trailing slash) or the URL of your Terraform Enterprise instance. The value for "client ID" is your audience in OIDC terminology and it should match the value of the `TFC_KUBERNETES_WORKLOAD_IDENTITY_AUDIENCE` environment variable in your workspace.
```
[the remark this answers is not in the document]
Comment (alexsomesan) [reply website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx]: The bolding of the word "without" was a copy-paste from the page about AWS dynamic credentials. Might be worth a look there too.

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx (hunk 5 above)
Comment (alexsomesan) [reply website/docs/cloud-docs/workspaces/dynamic-provider-credentials/kubernetes-configuration.mdx]: Indeed, that was copy pasta. I should have replaced ""AWS role" with "audience". Will correct it.

**Comment by alexsomesan** [comment]
Thank you so much for the review and corrections @rkoron007 and @BrianMMcClain!

Just a small observation from my side. Some of the suggested corrections were on fragments copied verbatim from the AWS version of the guide. It might be worth another look at that one too.

**Comment by rkoron007** [comment]
> Just a small observation from my side. Some of the suggested corrections were on fragments copied verbatim from the AWS version of the guide. It might be worth another look at that one too.

Oooh thank you for raising this! I'll make a note for myself to check the AWS guide out too 👀
